lib/common_test/src/ct.erl

```diff
@@ -89,34 +92,135 @@
 %%----------------------------------------------------------------------
 
 
+-spec install(Opts) -> ok | {error, Reason}
+      when Opts :: [Opt],
+           Opt :: {config, ConfigFiles} | {event_handler, Modules} | {decrypt, KeyOrFile},
+           ConfigFiles :: [ConfigFile],
+           ConfigFile :: string(),
+           Modules :: [atom()],
+           KeyOrFile :: {key, Key} | {file, KeyFile},
+           Key :: string(),
+           KeyFile :: string(),
+           Reason :: term().
 install(Opts) ->
     ct_run:install(Opts).
 
+-spec run(TestDir, Suite, Cases) -> Result
+      when TestDir :: string(),
+           Suite :: atom(),
+           Cases :: atom() | [atom()],
+           Result :: [TestResult] | {error, Reason},
+           TestResult :: term(),
+           Reason :: term().
 run(TestDir,Suite,Cases) ->
     ct_run:run(TestDir,Suite,Cases).
 
+-spec run(TestDir, Suite) -> Result
+      when TestDir :: string(),
+           Suite :: atom(),
+           Result :: [TestResult] | {error, Reason},
+           TestResult :: term(),
+           Reason :: term().
 run(TestDir,Suite) ->
     ct_run:run(TestDir,Suite).
 
+-spec run(TestDirs) -> Result
+      when TestDirs :: TestDir | [TestDir],
```

Comment:
I think the type variable `TestDir` should be `TestDir :: string()`

Reply:
2d6d21f01d